pkg/ee/cluster-backup/resources/seed-cluster/deployment.go

```diff
@@ -0,0 +1,238 @@
+	clusterBackupName                 = "velero-cluster-backup"
+	clusterbackupKubeConfigSecretName = "velero-kubeconfig"
+	cloudCredentialsSecretName        = "velero-cloud-credentials"
+
+	veleroImage = "velero/velero:v1.12.0"
+)
+
+// DeploymentReconciler creates the velero deployment in the user cluster namespace.
+func DeploymentReconciler(data *resources.TemplateData) reconciling.NamedDeploymentReconcilerFactory {
+	return func() (string, reconciling.DeploymentReconciler) {
+		return clusterBackupName, func(dep *appsv1.Deployment) (*appsv1.Deployment, error) {
+			dep.Labels = resources.BaseAppLabels(clusterBackupName, nil)
+
+			dep.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: resources.BaseAppLabels(clusterBackupName, nil),
+			}
+			dep.Spec.Replicas = resources.Int32(1)
+
+			dep.Spec.Template.Spec.ImagePullSecrets = []corev1.LocalObjectReference{{Name: resources.ImagePullSecretName}}
+
+			volumes := getVolumes()
+			volumeMounts := getVolumeMounts()
+
+			podLabels, err := data.GetPodTemplateLabels(clusterBackupName, volumes, nil)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create pod labels: %w", err)
+			}
+
+			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: podLabels,
+				Annotations: map[string]string{
+					"prometheus.io/path":   "/metrics",
+					"prometheus.io/port":   "8085",
+					"prometheus.io/scrape": "true",
+				},
+			}
+
+			dep.Spec.Template.Spec.Volumes = volumes
+			dep.Spec.Template.Spec.InitContainers = []corev1.Container{
+				{
+					Name:  "velero-velero-plugin-for-aws",
+					Image: "velero/velero-plugin-for-aws:v1.0.0",
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "plugins",
+							MountPath: "/target",
+						},
+					},
+					ImagePullPolicy: corev1.PullPolicy("IfNotPresent"),
+				},
+			}
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				{
+					Name:    "velero",
+					Image:   veleroImage,
+					Command: []string{"/velero"},
+					Args: []string{
+						"server",
+						"--features=",
+						"--uploader-type=kopia",
+						fmt.Sprintf("--namespace=%s", resources.ClusterBackupNamespaceName),
+						"--kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig",
+					},
+					Ports: []corev1.ContainerPort{
+						{
+							Name:          "metrics",
+							ContainerPort: 8085,
+							Protocol:      corev1.ProtocolSCTP,
+						},
+					},
+					Env: []corev1.EnvVar{
+						{
+							Name:  "VELERO_SCRATCH_DIR",
+							Value: "/scratch",
+						},
+						{
+							Name:  "LD_LIBRARY_PATH",
+							Value: "/plugins",
+						},
+						{
+							Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+							Value: "/credentials/cloud",
+						},
+						{
+							Name:  "AWS_SHARED_CREDENTIALS_FILE",
+							Value: "/credentials/cloud",
+						},
+						{
+							Name:  "AZURE_CREDENTIALS_FILE",
+							Value: "/credentials/cloud",
+						},
+						{
+							Name:  "ALIBABA_CLOUD_CREDENTIALS_FILE",
+							Value: "/credentials/cloud",
+						},
```

Comment:
We only load the AWS plugin, so why do we define other cloud provider credentials?

If Velero integration gated somehow, so it does not get enabled on incompatible providers?

Reply:
Auto generated by Velero, removed for now.